### PR TITLE
Tim   indicator relationships analysis

### PR DESCRIPTION
**Pull request by ArikDay**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4942)

## Description
This playbooks allows you to analyze indicator relationships for investigation purposes. 

## Screenshots
TIM_-_Indicator_Relationships_Analysis

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "16"
+    scriptarguments:
+      key:
+        simple: RelatedCampaign
+      value:
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:608]: add unique here as well

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 2]
```diff
@@ -0,0 +1,933 @@
...
+      key:
+        simple: RelatedReport
+      value:
+        complex:
+          root: Relationships
+          filters:
+          - - operator: isEqualString
+              left:
+                value:
+                  simple: Relationships.EntityBType
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:745]: should also check for the related campaigns you mentioned in task # 17?

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 3]
```diff
@@ -0,0 +1,933 @@
...
+  type: unknown
+- contextPath: RelatedIPs
+  description: IPs related to the indicator and campaign
+  type: unknown
+- contextPath: RelatedURLs
+  description: URLs related to the indicator and campaign
+  type: unknown
+tests:
+- No tests (auto formatted)
+fromversion: 6.5.0
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:933]: New content should be from 6.8

File: Packs/TIM_Processing/ReleaseNotes/1_1_13.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+#### Playbooks
+##### New: TIM - Indicator Relationships Analysis
+This playbooks allows you to analyze indicator relationships for investigation purposes. the playbook will provide the following relationships:
+- Related IOCs
+- Related Attack Patterns
+- Related Campaign
+- IOC's related to the campaign
+- Report that contains the campaign
+
+ (Available from Cortex XSOAR 6.5.0).
```
Comment (ssokolovich) [inline Packs/TIM_Processing/ReleaseNotes/1_1_13.md:11]: New content from 6.8

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 5]
```diff
@@ -0,0 +1,933 @@
...
+    isautoswitchedtoquietmode: false
+  "2":
+    id: "2"
+    taskid: 408729ca-ca33-4d44-8528-ba22c6afb08d
+    type: condition
+    task:
+      id: 408729ca-ca33-4d44-8528-ba22c6afb08d
+      version: -1
+      name: Has related  campaign?
+      description: check if the indicator related to campaign
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:52]: `Check if the indicator is related to a campaign.`

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 6]
```diff
@@ -0,0 +1,933 @@
...
+    isautoswitchedtoquietmode: false
+  "8":
+    id: "8"
+    taskid: 7a6cd9f0-7aed-43fb-8e6c-25e11dc4e1e2
+    type: condition
+    task:
+      id: 7a6cd9f0-7aed-43fb-8e6c-25e11dc4e1e2
+      version: -1
+      name: Has Files?
+      description: check if FILE type indicators exist
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:303]: `Check if File type indicator exists.`

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 7]
```diff
@@ -0,0 +1,933 @@
...
+    isautoswitchedtoquietmode: false
+  "9":
+    id: "9"
+    taskid: 81ba69cf-145c-4836-8fb7-ab3d856fd02b
+    type: condition
+    task:
+      id: 81ba69cf-145c-4836-8fb7-ab3d856fd02b
+      version: -1
+      name: Has URLs?
+      description: check if URL type indicators exist
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:350]: `Check if URL type indicator exists.`

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 8]
```diff
@@ -0,0 +1,933 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "8":
+    id: "8"
+    taskid: 7a6cd9f0-7aed-43fb-8e6c-25e11dc4e1e2
+    type: condition
+    task:
+      id: 7a6cd9f0-7aed-43fb-8e6c-25e11dc4e1e2
+      version: -1
+      name: Has Files?
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:302]: Is there a File type indicator?

**Review by ssokolovich — COMMENTED** [review]
Please check the comments, some of them are suggestions ( such as task names). Also, note that we are trying to keep the convention of a single element ( so it's `Indicator` and not `indicators`, but again depends on the need). let me know if you have any questions.

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 9]
```diff
@@ -0,0 +1,933 @@
...
+- key: Indicator
+  value: {}
+  required: false
+  description: Input indicator to analyze his relationships
+  playbookInputQuery:
+- key: LimitResults
+  value:
+    simple: "200"
+  required: false
+  description: The number of results to return. Default is 20.
```
Comment (ssokolovich) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:907]: Please change the description.

**Review by ssokolovich — APPROVED** [review]

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis_README.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+### Commands
+This playbook does not use any commands.
+
+## Playbook Inputs
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| Indicator | Input indicator to analyze its relationships. |  | Optional |
+| LimitResults | The number of results to return. If the input is empty, the limit will be 20. | 200 | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis_README.md:30]: If the limit is empty, then won't the default be used?  The default is 200, not 20. Or do I misunderstand?

File: Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml [hunk 11]
```diff
@@ -0,0 +1,942 @@
...
+- key: Indicator
+  value: {}
+  required: false
+  description: Input an indicator to analyze its relationships.
+  playbookInputQuery:
+- key: LimitResults
+  value:
+    simple: "200"
+  required: false
+  description: The number of results to return. If the input is empty, the limit will be 20.
```
Comment (ShirleyDenkberg) [inline Packs/TIM_Processing/Playbooks/playbook-TIM_-_Indicator_Relationships_Analysis.yml:916]: Does this mean that the default is 20?

**Comment by ShirleyDenkberg** [comment]
@ssokolovich Doc review completed.

**Pull request merged**